### ISSUE TITLE
[BUG] Does not recognizes csv fields with numbers

### ISSUE THREAD

**Issue by onlycparra** [siefkenj/MailMerge#26]
my csv:
```
email,name,score,comments
"a@b.c","John",10.8,"<ul><li>comment 1</li></ul>"
```

My email:
```
to: {{email}}
subj: some title
body:
Hello {{name}}, you got {{score}} in this test.
Comments: 
{{comments}}
Best,
Me
```
I obtain the final body text: 

```
Hello John, you got in this test.
Comments
- comment 1
Best,
Me
```

I expect to get the number in the sentence "you got 10.8 in this test."

The problem exists even if the numbers are quoted, even if it is just an integer. More over, if I have two records in my csv file:
```
email,name,score,comments
"a@b.c","John",10.8,"<ul><li>comment 1</li></ul>"
"x@y.z","Juan", "a 10.8","<ul><li>comment 2</li></ul>"
```
I get the blank behavior for John, but the mail of Juan gets the correct sentence "you got a 10.8 in this test." This rules out being a problem somewhere else. It is an issue with the replacement of numbers.

Thanks

**Comment by siefkenj** [comment]
If you use an xlsx file, does it work?

**Comment by onlycparra** [comment]
I use Linux. I cannot produce xlsx files.

----------------------------------------

Apr 20, 2022 16:18:23 Jason Siefken ***@***.***>:

**Comment by siefkenj** [comment]
Any of the office suites can.

What about if you copy-and-paste the data into the spreadsheet?

**Comment by onlycparra** [comment]
Oh, I see. I'll try it later today.

----------------------------------------

Apr 20, 2022 16:49:36 Jason Siefken ***@***.***>:

**Comment by onlycparra** [comment]
Hello, @siefkenj.
### Version of the plugin:
- Mail Merge P v2.4.
- Last updated on April 20, 2022. 
- Obtained from the same thunderbird, from: https://addons.thunderbird.net/en-US/thunderbird/addon/mail-merge-p/?src=search

### What I have tried:

I did try with xlsx and ods files, both generated by libreoffice. Here my findings.
- I have completely killed the thunderbird process before running each experiment.
- The  "data" tab of MailMerge does show the correct numeric values in the corresponding column (this also happens with csv files).
- In the "settings" tabs, I have tried using both parsers "Nunjucks" and "Legacy". The final result is the same.
- The "preview" tab shows nothing in the place where the numeric variables should have been replaced.
- In one record of the xls file, instead of a number, I used a string (see example below). The string appears correctly replaced in the preview.
- If I export the mails anyway to the "Outbox", the actual emails match what is seen in the preview (including the error on replacing the numbers)
- EVEN if I put the number within quotation marks, the problem persists (see example below).

### Specific example:

I tried this file as csv, xls, xlsx and ods. The result is always the same:
```
email,val1,val2,val3
"decimal@a.com",1.33,2.18,3.0
"int@b.com",10,20,30
"names@c.com","one","two","three"
"quoted@d.com","100","200","300"
"spaced@e.com", 1, 2, 3
```

Template:
```
To: {{val1}} <{{email}}>
Subject: Some title
Body:
{{val1}}, {{val2}}, {{val3}}
Bye
```

Emails composed: 
```
------------------
To: <decimal@a.com>
Subject: Some title
Body:
, ,
Bye
------------------
To: <int@b.com>
Subject: Some title
Body:
, ,
Bye
------------------
To: one <names@c.com>
subject: Some title
Body:
one, two, three
Bye
------------------
To: <quoted@d.com>
subject: Some title
Body:
, , 
Bye
------------------
To: <spaced@e.com>
subject: Some title
Body:
, , 
Bye
------------------
```

I hope this helps.

**Comment by onlycparra** [comment]
any news on this?

**Comment by siefkenj** [comment]
@onlycparra This should now be fixed in version 2.5!

**Comment by onlycparra** [comment]
Awesome, I just tested it with some integers. It seems to be working well. Thanks. The next coffee is on me :)
